### PR TITLE
Add a user-agent when deploying with appcfg

### PR DESCRIPTION
**Pull request by nbashirbello**
@loosebazooka @patflynn @elharo

### CODE REVIEW

File: core-plugin/src/com/google/gct/idea/appengine/deploy/UserAgentStrings.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * it should be reflected in the name of the user-agent e.g 'MAVEN_VIA_CLOUD_TOOLS_FOR_ANDROID_STUDIO'
+ */
+public class UserAgentStrings {
+    public static String USER_AGENT_KEY = "appengine.useragent";
+
+    private UserAgentStrings () {
+    }
+
+    // Sources for appcfg update/deploy calls
+    public static String CT4AS = "CLOUD_TOOLS_FOR_ANDROID_STUDIO";
```
Comment (elharo) [inline core-plugin/src/com/google/gct/idea/appengine/deploy/UserAgentStrings.java]: ANDROID_STUDIO?

**Comment by patflynn** [comment]
@nbashirbello @elharo @loosebazooka what do you think about moving the AE1 deploy codebase into our Android Studio plugin?

File: core-plugin/src/com/google/gct/idea/appengine/deploy/UserAgentStrings.java (hunk 1 above)
Comment (nbashirbello) [inline core-plugin/src/com/google/gct/idea/appengine/deploy/UserAgentStrings.java]: Done

**Comment by elharo** [comment]
If we do thhat can it also be deleted from whereever it lives now?

**Comment by patflynn** [comment]
yes. that would be the idea.

**Comment by elharo** [comment]
I don't have any objections then. 

The only issue I can imagine is that it might be a tad confusing to external devs since we don't support deploying from inside IntelliJ, but that's not major.

**Comment by nbashirbello** [comment]
Created an issue for moving the AE1 code to Andriod Studio: https://github.com/GoogleCloudPlatform/gcloud-intellij/issues/199

**Comment by elharo** [comment]
lgtm

**Pull request merged**